aptos-move/framework/aptos-framework/sources/dora_committee.move

```diff
@@ -0,0 +1,1048 @@
+        resource_signer
+    }
+
+    /// Its Initial function which will be executed automatically while deployed packages
+    fun init_module(owner_signer: &signer) {
+        create_owner(owner_signer);
+        create_committeeInfo_store(owner_signer);
+        create_event_handler(owner_signer);
+    }
+
+    #[view]
+    /// Get the committee's dora node vector
+    public fun get_committee_info(com_store_addr: address, id: u64): vector<NodeData> acquires CommitteeInfoStore {
+        let committee_store = borrow_global<CommitteeInfoStore>(com_store_addr);
+        let committee = simple_map::borrow(&committee_store.committee_map, &id);
+        let (addrs, dora_nodes) = simple_map::to_vec_pair(committee.map);
+        let node_data_vec = vector::empty<NodeData>();
+        while (vector::length(&addrs) > 0) {
+            let addr = vector::pop_back(&mut addrs);
+            let dora_node_info = vector::pop_back(&mut dora_nodes);
+            let node_data = NodeData {
+                operator: addr,
+                ip_public_address: dora_node_info.ip_public_address,
+                dora_public_key: dora_node_info.dora_public_key,
+                network_public_key: dora_node_info.network_public_key,
+                elgamal_pub_key: dora_node_info.elgamal_pub_key,
+                network_port: dora_node_info.network_port,
+                rpc_port: dora_node_info.rpc_port,
+            };
+            vector::push_back(&mut node_data_vec, node_data);
+        };
+        node_data_vec
+    }
+
+    #[view]
+    /// Get the committee's ids from the store
+    public fun get_committee_ids(com_store_addr: address): vector<u64> acquires CommitteeInfoStore {
+        let committee_store = borrow_global<CommitteeInfoStore>(com_store_addr);
+        simple_map::keys(&committee_store.committee_map)
+    }
+
+    #[view]
+    /// Get the committee's id for a single node, only pass the address is okay
+    public fun get_committee_id(
+        com_store_addr: address,
+        map_key: vector<address>
+    ): u64 acquires CommitteeInfoStore {
+        let committee_store = borrow_global<CommitteeInfoStore>(com_store_addr);
+        let node_address = vector::pop_back(&mut map_key);
+        let id = *simple_map::borrow(&committee_store.node_to_committee_map, &node_address);
+        id
+    }
+
+    #[view]
+    /// Get the node's information
+    public fun get_node_info(
+        com_store_addr: address,
+        id: u64,
+        node_address: address
+    ): NodeData acquires CommitteeInfoStore {
+        let committee_store = borrow_global<CommitteeInfoStore>(com_store_addr);
+        let committee = simple_map::borrow(&committee_store.committee_map, &id);
+        let (addrs, dora_nodes) = simple_map::to_vec_pair(committee.map);
+        let (flag, index) = vector::index_of(&addrs, &node_address);
+        let dora_node_info = vector::borrow(&dora_nodes, index);
+        assert!(flag, error::invalid_argument(NODE_NOT_FOUND));
```

Comment:
```suggestion
        assert!(flag, error::invalid_argument(NODE_NOT_FOUND));
        let dora_node_info = vector::borrow(&dora_nodes, index);
  
```
`assert` should come before we access the `index` otherwise this may `panic`.